thanos/charts/alerts/query.yaml

```
@@ -0,0 +1,147 @@
+      {{ tpl .Values.thanos.serviceMonitor.alertLabels . | nindent 6 }}
+  - alert: ThanosQueryInstantLatencyHigh
+    annotations:
+      description: Thanos Query `{{`{{$labels.job}}`}}` has a 99th percentile latency of `{{`{{$value}}`}}`
+        seconds for instant queries.
+      summary: Thanos Query has high latency for queries.
+    expr: |
+      (
+        histogram_quantile(0.99, sum by (job, le) (rate(http_request_duration_seconds_bucket{job=~".*query.*", handler="query"}[5m]))) > 40
+      and
+        sum by (job) (rate(http_request_duration_seconds_bucket{job=~".*query.*", handler="query"}[5m])) > 0
+      )
+    for: 10m
+    labels:
+      severity: critical
+      playbook: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanosqueryinstantlatencyhigh
+      {{ tpl .Values.thanos.serviceMonitor.alertLabels . | nindent 6 }}
+  - alert: ThanosQueryRangeLatencyHigh
+    annotations:
+      description: Thanos Query `{{`{{$labels.job}}`}}` has a 99th percentile latency of `{{`{{$value}}`}}`
+        seconds for range queries.
+      summary: Thanos Query has high latency for queries.
+    expr: |
+      (
+        histogram_quantile(0.99, sum by (job, le) (rate(http_request_duration_seconds_bucket{job=~".*query.*", handler="query_range"}[5m]))) > 90
+      and
+        sum by (job) (rate(http_request_duration_seconds_count{job=~".*query.*", handler="query_range"}[5m])) > 0
+      )
+    for: 10m
+    labels:
+      severity: critical
+      playbook: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanosqueryrangelatencyhigh
+      {{ tpl .Values.thanos.serviceMonitor.alertLabels . | nindent 6 }}
+  - alert: ThanosQueryOverload
+    annotations:
+      description: Thanos Query `{{`{{$labels.job}}`}}` has been overloaded for more than
+        15 minutes. This may be a symptom of excessive simultaneous complex requests,
+        low performance of the Prometheus API, or failures within these components.
+        Assess the health of the Thanos query instances, the connected Prometheus
+        instances, look for potential senders of these requests and then contact support.
+      summary: Thanos query reaches its maximum capacity serving concurrent requests.
+    expr: |
+      (
+        max_over_time(thanos_query_concurrent_gate_queries_max[5m]) - avg_over_time(thanos_query_concurrent_gate_queries_in_flight[5m]) < 1
+      )
+    for: 15m
+    labels:
+      severity: warning
+      playbook: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanosqueryoverload
+      {{ tpl .Values.thanos.serviceMonitor.alertLabels . | nindent 6 }}
+- name: thanos-component-absent
+  rules:
+  - alert: ThanosQueryIsDown
+    annotations:
+      description: ThanosQuery has disappeared. Prometheus target for the component
+        cannot be discovered.
+      summary: Thanos component has disappeared.
+    expr: |
+      absent(up{job=~".*query.*"} == 1)
+    for: 5m
```

```suggestion
    for: 10m
```

would give this more slack